### ISSUE TITLE
Add support for font kerning

### ISSUE THREAD

**Issue by virtualritz** [nodebox/nodebox#358]
It seems Nodebox ignores any kerning information stored in a font. It looks like characters are just positioned by their bounding boxes.

This makes Nodebox pretty much unusable for any typographic work with the least considerations for aesthetics.

**Comment by wakest** [comment]
getting stuck on this too. is there any fix for this on OS X yet?

**Comment by fdb** [comment]
Here's the line that enables kerning support:

https://github.com/nodebox/nodebox/blob/2717b34a7fa57f64282bcc2e4f07ed65ac2b1d25/src/main/java/nodebox/graphics/Text.java#L134

We're now using the latest version of the SDK (1.8) on Mac, which should take kerning information into account. (since it's the same version as on Windows). @wakest Are you using the latest version?

**Comment by wakest** [comment]
@fdb I tried to use the latest version but it instantly crashes on startup and gives leaves a .log file in my home directory:
```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fff8f525152, pid=585, tid=4879
#
# JRE version: Java(TM) SE Runtime Environment (8.0_25-b17) (build 1.8.0_25-b17)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.25-b02 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libsystem_c.dylib+0x1152]  strlen+0x12
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
```

**Comment by fdb** [comment]
Super-annoying, since I can't reproduce this on my machine.

Could you try to do what it says? Go to the terminal, execute `ulimit -c unlimited`, then start the app again and retrieve the crash log, if it's available.

**Comment by fdb** [comment]
Also, what was the version you were using before that *didn't* crash?

**Comment by wakest** [comment]
tried executing '''ulimit -c unlimited''' a few times, always the same error log. crashes instantly on startup. I am using 3.0.40 which works great and I am super excited to be playing with, just wish I could get kerning support from the new version.

I saw you mentioned if built though homebrew it should work because it uses the system java, did I understand that correctly?

heres the full crash report file and the log file from 3.0.43: https://www.dropbox.com/sh/dkvmei1bfl5p8bu/AAAqUFJsZCwAJN2h7WQF2c2ma?dl=0

**Comment by fdb** [comment]
Thanks for emailing me the crash reports. That's very useful.

You mentioned you're using 3.0.40 now. After that version, we started bundling our own Java to support Yosemite. Apparently the bundling is problematic. (And apparently [Processing has the same issue](https://github.com/processing/processing/issues/3360))

You can always run from the terminal if you want the latest and greatest, and even build your own versions.


    git clone https://github.com/nodebox/nodebox.git
    cd nodebox
    ant dist-mac
    open dist

This will open the dist folder and it will contain your own version of NodeBox, that's built from the latest source. You might have to do `brew update && brew install ant maven` if you don't have those tools yet.

Meanwhile, I'll try creating a newer version that bundles a more up-to-date version of Java.

**Comment by fdb** [comment]
@wakest I've moved the crashing to a separate issue: #407. Presumably, if we fix that, you'll be able to open the most recent version and it will have good kerning support.
